rust/kernel/cpumask.rs

```diff
@@ -177,12 +181,92 @@ impl CpumaskT {
         &mut self.0 as *mut bindings::cpumask_t
     }
 
+    /// The `as_cpumas_const_ptr` method returns a const pointer to the underlying `bindings::cpumask_t`. This can be used to pass the `CpumaskT` to kernel functions that expect a `bindings::cpumask_t`.
+    pub fn as_cpumas_const_ptr(&self) -> *const bindings::cpumask_t {
+        &self.0 as *const bindings::cpumask_t
+    }
+
     /// The `cpu_mask_all` method returns a `CpumaskT` that represents all CPUs. It does this by creating a new `CpumaskT` with all bits set to 1.
-    pub fn cpu_mask_all() -> Self {
+    pub const fn cpu_mask_all() -> Self {
         let c: u64 = u64::MAX;
         Self(bindings::cpumask_t { bits: [c, c, c, c] })
         // Self(bindings::cpumask_t { bits: [c] })
     }
+
+    /// The `read_cpu_online_mask` function is an unsafe function that reads the CPU online mask from the kernel. It returns a `bindings::cpumask` representing the online CPUs.
+    pub unsafe fn read_cpu_online_mask() -> Self {
+        unsafe {
+            Self {
+                0: bindings::__cpu_online_mask,
+            }
+        }
+    }
```

Comment:
Reading global varibale needs a lock. I remeber `online cpu` has a correspending lock.

Reply:
Sorry, I didn't find the corresponding lock. Can you tell me where it is?

By the way, does `RROS` support `CPU` hotplug? I didn't find any code in `RROS` and `EVL` to support `CPU` hotplug.

Reply:
The lock is `cpus_read_lock`.

RROS does not support this. But the abstraction stays in the rfl layer. We still need to keep it safe.

Reply:
Ok, thanks.